src/python/genny/genny_auto_tasks.py

```diff
@@ -45,17 +80,83 @@ def modified_workload_files():
     return short_filenames
 
 
-def get_project_root():
+def workload_should_autorun(workload_dict, env_dict):
     """
-    :return: the path of the project root.
+    Check if the given workload's AutoRun conditions are met by the current environment
+    :param dict workload_dict: a dict representation of the workload files's yaml.
+    :param dict env_dict: a dict representing the values from bootstrap.yml and runtime.yml
+    :return: True if this workload should be autorun, else False.
     """
-    try:
-        out = subprocess.check_output('git rev-parse --show-toplevel', shell=True)
-    except subprocess.CalledProcessError as e:
-        print(e.output, file=sys.stderr)
-        raise e
 
-    return out.decode().strip()
+    # First check that the workload has a proper AutoRun section.
+    if 'AutoRun' not in workload_dict or not isinstance(workload_dict['AutoRun'], dict):
+        return False
+    if 'Requires' not in workload_dict['AutoRun'] or not isinstance(workload_dict['AutoRun']['Requires'], dict):
+        return False
+
+    for module, required_config in workload_dict['AutoRun']['Requires'].items():
+        if module not in env_dict:
+            return False
+        if not isinstance(required_config, dict):
+            return False
+
+        # True if set of config key-vaue pairs is subset of env_dict key-value pairs
+        # This will be false if the AutoRun yaml uses a logical or (i.e. branch_name: master | prod), but it is efficient so we use it for a first-pass.
+        if not required_config.items() <= env_dict[module].items():
```

Comment:
I'm debating between using "|" or a list for multiple environment variables. Any preference @rtimmons?

Reply:
I chatted with @rtimmons in passing yesterday and I think he prefers lists. String parsing is always a bit magical and the "|" convention may not seem intuitive for Windows folks

Reply:
I agree with that logic and changed the code accordingly